relay-server/src/services/metrics/aggregator.rs

```diff
@@ -135,7 +155,47 @@ impl AggregatorService {
     /// If `force` is true, flush all buckets unconditionally and do not attempt to merge back.
     fn try_flush(&mut self) {
         let force_flush = matches!(&self.state, AggregatorState::ShuttingDown);
-        let partitions = self.aggregator.pop_flush_buckets(force_flush);
+
+        let partitions = self.aggregator.pop_flush_buckets(
+            force_flush,
+            |project_key| {
```

Comment:
nit: Worth assigning this closure to a variable to clarify that it is the flush decider?